numpy/linalg/linalg.py

```diff
@@ -241,7 +241,7 @@ def transpose(a):
     -------
     aT : (...,N,M) ndarray
     """
-    return swapaxes(a, -1, -2)
+    return swapaxes.__skip_array_function__(a, -1, -2)
```

Comment:
The transpose here is not actually decorated as a skipping function, because it is linalg only. So I am really unsure that this is correct. (very likely my ignorance though).

Reply:
I *think* this is right, because this is always used inside other linalg functions, which *are* all wrapped with `__array_function__` and also cast their inputs into NumPy arrays (see the `_make_array` utility function). And even if we had some functions that *didn't* do this, we don't want to expose the fact that they happen to use `swapaxes` internally. That would be leaking an implementation detail.

Reply:
OK, this is whole thing is new anyway. The function is just publicly exposed, so that means if someone actually uses it, it will fail on some array likes (and it would not before).

Reply:
Is `transpose` really publicly exposed? I guess as `np.linalg.linalg.transpose`, but it isn't documented anywhere.

Reply:
But to be clear, if somebody used it on array-likes before, this function would indeed really work the same. There hasn't been a release of NumPy that enables `__array_function__` yet.

Reply:
`linalg.linalg` isn't public

Reply:
Oh, sorry, I had missed the second linalg there somehow...

Reply:
> we don't want to expose the fact that they happen to use swapaxes internally. That would be leaking an implementation detail.

Calling it an inplementation seems like a stretch to me. The 2d transpose is essentially defined as swapping the last two axes. If linalg.transpose were public, then I'd argue that it should remain unchanged.

As written here, it seems like a questionable but harmless micro-optimization.